### PR TITLE
enable more test cases for ci.yml

### PR DESCRIPTION
**Pull request by chobits**
1. enable ssl module for ngx_dtls.t (this case is skipped before)
2. enable addition module for http_raw_uri_variable.t
3. fixed ngx_dtls.t: 1. too small certificate key error 2. check verification string with case insensitivity
5. fixed segfault found by ngx_dtls.t, for details see here: https://github.com/alibaba/tengine/pull/1205#discussion_r1019963385

---

With this pr, all test cases in [tests/nginx-tests/tengine-tests](https://github.com/alibaba/tengine/tree/master/tests/nginx-tests/tengine-tests) will be checked.

### CODE REVIEW

**Comment by chobits** [comment]
aslo cc @u5surf for ci.yml modified

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -43,7 +43,40 @@ jobs:
...
+         ./configure \
+            --with-ld-opt="-Wl,-rpath,/usr/local/lib" \
+            --with-ipv6 \
+            --with-http_ssl_module \
+            --with-http_v2_module \
+            --with-http_addition_module \
```
Comment (chobits) [inline .github/workflows/ci.yml:51]: new compiled addtion module

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -43,7 +43,40 @@ jobs:
...
+            --with-stream \
+            --with-stream_ssl_module \
+            --with-stream_realip_module \
+            --with-stream_geoip_module \
+            --with-stream_ssl_preread_module \
+            --with-stream_sni \
```
Comment (chobits) [inline .github/workflows/ci.yml:57]: new compiled stream and stream_* module

File: src/stream/ngx_stream_ssl_module.c [hunk 3]
```diff
@@ -875,7 +875,6 @@ ngx_stream_ssl_merge_conf(ngx_conf_t *cf, void *parent, void *child)
 
 #if (NGX_STREAM_SNI)
     ngx_conf_merge_value(conf->sni_force, prev->sni_force, 0);
-    if (!conf->listen)
```
Comment (chobits) [inline src/stream/ngx_stream_ssl_module.c:878]: this line trigger segfault

File: tests/nginx-tests/tengine-tests/ngx_dtls.t [hunk 4]
```diff
@@ -45,7 +45,7 @@ EOF
 
 $t->write_file('openssl.conf', <<EOF);
 [ req ]
-default_bits = 1024
+default_bits = 2048
```
Comment (chobits) [inline tests/nginx-tests/tengine-tests/ngx_dtls.t:48]: ssl certificate key too small error

File: tests/nginx-tests/tengine-tests/ngx_dtls.t [hunk 5]
```diff
@@ -63,7 +63,8 @@ foreach my $name ('localhost') {
 
 
 $t->run();
-my $ret1 = `openssl s_client -connect 127.0.0.1:8980 -dtls1 | grep "ok"`;
+my $ret1 = `openssl s_client -connect 127.0.0.1:8980 -dtls1 | grep -i "ok"`;
```
Comment (chobits) [inline tests/nginx-tests/tengine-tests/ngx_dtls.t:66]: ignore case of string checking

**Comment by chobits** [comment]
CI/CD passed, see https://github.com/alibaba/tengine/actions/runs/3443659256/jobs/5745429941

**Review by wa5i — APPROVED** [review]

**Review by wangfakang — APPROVED** [review]

**Pull request merged**